cni/pkg/install/cniconfig_test.go

```diff
@@ -21,10 +21,11 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
```

Comment:
Yeah I was gonna ask about this, actually.

I know istio has `"istio.io/istio/pkg/test/util/assert"`

`testify`'s assert is a drop-in replacement for `"istio.io/istio/pkg/test/util/assert"`, supports more things, and comes with a very nice mocking library too.

I kinda like it better? And I dunno if we really need our own in-tree assert lib given that `testify` is pretty popular and does a better job.

There may be historical context I don't have on this, but I'd as soon replace all uses of `"istio.io/istio/pkg/test/util/assert"` with testify, and get both better asserts and good dynamic mocking (rather than rolling DIY mocks which seems to be current default).

Not to mention we're already using `testify/mock` so testify is being pulled in anyway.

Reply:
Its not a drop in, it has different semantics which will break most Istio usages. Also it exposes a ton of junk; we explicitly have only 2-3 assertions instead of the junk like `assert.Is().GreaterThan(3).Or(...)` like other libraries do (I don't think testify is that bad, but still).

Mock is orthogonal, and per my other comment I am strongly against

Reply:
The semantic differences for the existing tests in this code were very, very minor, but I don't much care personally as long as we keep the mocking bit, so that's fine with me.